plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsVmExecutorService.java

```diff
@@ -77,9 +81,11 @@ public Void call() {
 
     @Override
     protected Runnable wrap(final Runnable r) {
+        Metrics.metricRegistry().counter(QUEUE_SIZE_METRIC).inc();
         return () -> {
+            Metrics.metricRegistry().counter(QUEUE_SIZE_METRIC).dec();
```

Comment:
I also considered using a `Timer` here instead of a `Counter` to be able to see how long tasks spend in the queue, but the problem is that `Timer`s only get counted when they are closed/stopped, and in the cases I have seen recently, the whole issue was that tasks were sitting in the queue for a very long time, so we need a metric that shows the problem while it is happening, not just after the blockage has been cleared.

This is also why I am not sure about trying to use `CpsFlowExecution.time` to track this data, but we could probably make it work with some API adjustments.

We could also consider using _both_ a `Counter` and a `Timer`.